controllers/datadogagent/merger/cilium.go

```diff
@@ -0,0 +1,591 @@
+}
+
+func (m *ciliumPolicyManagerImpl) SetupCiliumManager(site, ddURL string, hostNetwork bool, dnsSelectorEndpoints []metav1.LabelSelector) {
+	m.site = site
+	m.ddURL = ddURL
+	m.hostNetwork = hostNetwork
+	m.dnsSelectorEndpoints = dnsSelectorEndpoints
+}
+
+// AddCiliumPolicies creates or updates multiple cilium network policies
+func (m *ciliumPolicyManagerImpl) CreateCiliumPolicy(name, namespace string, policySpecs []cilium.NetworkPolicySpec) error {
+	obj, _ := m.store.GetOrCreate(kubernetes.CiliumNetworkPoliciesKind, namespace, name)
+	_, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("unable to get from the store the CiliumPolicy %s", name)
+	}
+
+	newPolicy := cilium.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Specs: policySpecs,
+	}
+
+	unstructured := &unstructured.Unstructured{}
+	var err error
+	unstructured.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(newPolicy)
+	if err != nil {
+		return err
+	}
+	unstructured.SetGroupVersionKind(cilium.GroupVersionCiliumNetworkPolicyKind())
+	return m.store.AddOrUpdate(kubernetes.CiliumNetworkPoliciesKind, unstructured)
+}
+
+// BuildCiliumPolicy creates the base node agent, DCA, or CCR cilium network policy
+func (m *ciliumPolicyManagerImpl) BuildCiliumPolicy(dda metav1.Object, componentName v2alpha1.ComponentName) error {
+	policyName, podSelector := getPolicyMetadata(dda, componentName)
+	var policySpecs []cilium.NetworkPolicySpec
+
+	switch componentName {
+	case v2alpha1.NodeAgentComponentName:
+		policySpecs = []cilium.NetworkPolicySpec{
+			egressECSPorts(podSelector),
+			egressNTP(podSelector),
+			egressMetadataServerRule(podSelector),
+			egressDNS(podSelector, m.dnsSelectorEndpoints),
+			egressAgentDatadogIntake(podSelector, m.site, m.ddURL),
+			egressKubelet(podSelector),
+			ingressDogstatsd(podSelector),
+			egressChecks(podSelector),
+		}
+	case v2alpha1.ClusterAgentComponentName:
+		_, nodeAgentPodSelector := getPolicyMetadata(dda, v2alpha1.NodeAgentComponentName)
+		policySpecs = []cilium.NetworkPolicySpec{
+			egressMetadataServerRule(podSelector),
+			egressDNS(podSelector, m.dnsSelectorEndpoints),
+			egressDCADatadogIntake(podSelector, m.site, m.ddURL),
+			egressKubeAPIServer(),
+			ingressAgent(podSelector, dda, m.hostNetwork),
+			ingressDCA(podSelector, nodeAgentPodSelector),
+			egressDCA(podSelector, nodeAgentPodSelector),
+		}
+	case v2alpha1.ClusterChecksRunnerComponentName:
+		policySpecs = []cilium.NetworkPolicySpec{
+			egressMetadataServerRule(podSelector),
+			egressDNS(podSelector, m.dnsSelectorEndpoints),
+			egressCCRDatadogIntake(podSelector, m.site, m.ddURL),
+			egressCCRToDCA(podSelector, dda),
+			egressChecks(podSelector),
+		}
+	}
+	return m.CreateCiliumPolicy(policyName, dda.GetNamespace(), policySpecs)
+}
+
+func getPolicyMetadata(dda metav1.Object, componentName v2alpha1.ComponentName) (policyName string, podSelector metav1.LabelSelector) {
```

Comment:
if i'm reading correctly, this is identical to the function of the same name in merger/network_policy.go. should we move to a common file?